src/intersections/segments.jl

```diff
@@ -1,161 +1,64 @@
 # ------------------------------------------------------------------
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
-
 #=
-The intersection type can be one of five types according to
-Balbes, R. and Siegel, J. 1990:
-
-1. intersect at point interior to both segments
-2. intersect at end point of one and an interior point of the other
-3. intersect at end point of both segments
-4. overlap at more than one point
-5. do not overlap nor intersect
-
-Balbes, R. and Siegel, J. 1990.
-(https://www.sciencedirect.com/science/article/abs/pii/0167839691900198)
+The intersection type can be one of six types:
+1. intersect at one inner point (CrossingSegments -> Point)
+2. intersect at one endpoint of one segment (MidTouchingSegments -> Point)
+3. intersect at one endpoint of both segments (CornerTouchingSegments -> Point)
+4. overlap of segments (OverlappingSegments -> Segments)
+5. do not overlap nor intersect (NoIntersection)
 =#
-function intersection(f, s1::Segment{2,T}, s2::Segment{2,T}) where {T}
-  x̄ = centroid(s1)
-  ȳ = centroid(s2)
-  x1, x2 = vertices(s1)
-  y1, y2 = vertices(s2)
-  Q = [x1, y1, x2, y2, x1]
-
-  # winding number of x̄
-  windingx = zero(T)
-  determinatex = true
-  for j in 1:length(Q)-1
-    α = ∠(Q[j], x̄, Q[j+1])
-    if α ≈ 0 || α ≈ π || α ≈ -π
-      determinatex = false
-    end
-    windingx += α
-  end
-
-  # winding number of ȳ
-  windingy = zero(T)
-  determinatey = true
-  for j in 1:length(Q)-1
-    β = ∠(Q[j], ȳ, Q[j+1])
-    if β ≈ 0 || β ≈ π || β ≈ -π
-      determinatey = false
-    end
-    windingy += β
-  end
-
-  # winding number ≈ 0?
-  wxzero = isapprox(windingx, zero(T), atol=atol(T))
-  wyzero = isapprox(windingy, zero(T), atol=atol(T))
-
-  # there are three cases to consider
-  # Balbes, R. and Siegel, J. 1990.
-  if determinatex && determinatey # CASE (I)
-    if !wxzero && !wyzero
-      # configuration (1)
-      return @IT CrossingSegments intersectpoint(s1, s2) f
+function intersection(f, s1::Segment{N,T}, s2::Segment{N,T}) where {N,T}
+  a, b = s1(0), s1(1)
+  c, d = s2(0), s2(1)
+
+  λ₁, λ₂, r, rₐ = intersectparameters(a, b, c, d)
+  # helper function to round values close to x (one or zero)
+  approx(λ, x) = isapprox(λ, x, atol=atol(T)) ? x : λ
+
+  if r ≠ rₐ # not in same plane or parallel
+    return @IT NoIntersection nothing f #CASE 5
+  elseif r == rₐ == 1 # collinear
+    # find parameters λc and λd for points c and d in s1
+    # use dimension with largest vector component to avoid division by zero
+    v₁ = b-a
+    i = argmax(abs.(v₁))
+    λc, λd = ((c-a)[i], (d-a)[i]) ./ v₁[i]
```

Comment:
please add space before and after `-` to improve readability

Reply:
@dorn-gerhard can you adjust this code style issue?

Reply:
there are multiple places where spaces can help readability

Reply:
i added more spaces